src/App.js

```diff
@@ -0,0 +1,108 @@
+import './App.css';
+import {Checklist} from './Checklist.js'
+import {AddTaskPage} from "./AddTaskPage";
+import {useEffect, useState} from "react";
+import {IncompleteTasksOnly} from "./IncompleteTasksOnly";
+import {generateUniqueID} from "web-vitals/dist/modules/lib/generateUniqueID";
+import {ModifyTaskPage} from "./ModifyTaskPage";
+
+export function App(props) {
+
+    let [data, setData] = useState(props.initialData);
+    let [itemsToShow, setItemsToShow] = useState("both");
+    const initialCompletedItems = data.map((item) => item.completed ? item.id : undefined);
+    let [completedItems, setCompletedItems] = useState(initialCompletedItems);
+    let [componentsToRender, setComponentsToRender] = useState(<Checklist items={data} handleChangeField={handleChangeField} modifyTask={modifyTask} completedItems={completedItems} changeCompletedItems={setCompletedItems}/>);
+
+    let taskButtonValue = (itemsToShow === "both" ? "Hide Completed Tasks" : "Show Completed Tasks");
+    let [showButtons, setShowButtons] = useState(true);
+
+    useEffect(() => {
```

Comment:
I don't understand why you're using useEffect here. Seems unnecessary.

Reply:
You should be able to rewrite this without using useEffect. Anytime you call a setState routine from a component, that'll cause that component to re-render.